scaffolding/src/main.rs

```diff
@@ -59,13 +60,36 @@ fn main() {
     env::set_var("TERM", "notty");
     let (tty_r, tty_w) = tty::pty("sh", COLS as u16, ROWS as u16);
 
-    // Handler program output (tty -> screen) on separate thread.
+    // Handle program output (tty -> screen) on separate thread.
     let (tx_out, rx) = mpsc::channel();
     let (tx_key_press, tx_key_release) = (tx_out.clone(), tx_out.clone());
+
+    let pty_open = Arc::new(AtomicBool::new(true));
+    let pty_open_checker = pty_open.clone();
     thread::spawn(move || {
         let output = Output::new(BufReader::new(tty_r));
-        for cmd in output {
-            tx_out.send(cmd.unwrap_or_else(exit_on_io_error)).unwrap();
+        for result in output {
+            match result {
+                Ok(cmd) => {
+                    tx_out.send(cmd).unwrap();
+                },
+                Err(_) => {
```

Comment:
Erh, I didn't get to the clarifying part. Not only is this the wrong way to terminate the application on systems that allow it, calling `gtk::main_quit()` from outside the GTK main thread is not guaranteed *to* terminate it.